File: client/src/api/invitations.ts
import { api } from './axios';
import type { UserInvitation } from '@/types/auth';
import type { ApiPaginated } from '@/types/api';

export const invitationApi = {
  list: (params: { page?: number; limit?: number } = {}) =>
    api
      .get<ApiPaginated<UserInvitation>>('/client/invitations', { params })
      .then((r) => r.data),

  resend: (id: string) =>
    api.post(`/client/invitations/${id}/resend`).then((r) => r.data.data),

  cancel: (id: string) => api.delete(`/client/invitations/${id}`).then((r) => r.data),
};

File: Android/client/lib/api/invitations.dart
import 'api_client.dart';

class InvitationsApi {
  static final _api = BizOsApiClient.instance;

  static Future<Response<dynamic>> list({Map<String, dynamic>? query}) =>
      _api.get('/client/invitations', query: query);
  static Future<Response<dynamic>> resend(String id) =>
      _api.post('/client/invitations/$id/resend');
  static Future<Response<dynamic>> cancel(String id) =>
      _api.delete('/client/invitations/$id');
}
